src/main/java/org/jabref/model/pdf/search/EnglishStemAnalyzer.java

```diff
@@ -0,0 +1,25 @@
+package org.jabref.model.pdf.search;
+
+import org.apache.lucene.analysis.Analyzer;
+import org.apache.lucene.analysis.LowerCaseFilter;
+import org.apache.lucene.analysis.StopFilter;
+import org.apache.lucene.analysis.TokenStream;
+import org.apache.lucene.analysis.Tokenizer;
+import org.apache.lucene.analysis.core.DecimalDigitFilter;
+import org.apache.lucene.analysis.en.EnglishAnalyzer;
+import org.apache.lucene.analysis.en.PorterStemFilter;
+import org.apache.lucene.analysis.standard.StandardTokenizer;
+
+public class EnglishStemAnalyzer extends Analyzer {
+
+    @Override
+    protected TokenStreamComponents createComponents(String fieldName) {
+        Tokenizer source = new StandardTokenizer();
+        TokenStream filter = new LowerCaseFilter(source);
+        filter = new StopFilter(filter, EnglishAnalyzer.ENGLISH_STOP_WORDS_SET);
+        filter = new DecimalDigitFilter(filter);
+        filter = new PorterStemFilter(filter);
```

Comment:
Looks strange. Maybe rename to stopFiler, decimalDigitFilter, porterStomFilter. Not really sure, what is easier to read.

Reply:
This is how people seem to do it. You basically chain them.
If you find it hard to read, instead of introducing a variable I would rather nest the calls.
`filter = new PorterStemFilter(new DecimalDigitFilter(new StopFilter(...)))`

Reply:
Meh... don't really have an opinion on this. maybe just leave it as it is.